qiskit/pulse/macros.py

```diff
@@ -47,7 +51,41 @@ def measure(
 
     Returns:
         A measurement schedule corresponding to the inputs provided.
+    """
+
+    # backend is V2.
+    if hasattr(backend, "target"):
+        return measure_v2(qubits, backend, target, meas_map, qubit_mem_slots, measure_name)
+    # backend is V1 or backend is None.
+    else:
+        return measure_v1(qubits, backend, inst_map, meas_map, qubit_mem_slots, measure_name)
+
 
+def measure_v1(
```

Comment:
This should be a protected method `_measure_v1` to minimize the maintenance overhead. Same for another function.